site/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,163 @@
+  mode?: TriggerMode;
+  defaultOpen?: boolean;
+}) => {
+  const [open, setOpen] = useState(props.defaultOpen ?? false);
+  const triggerRef = useRef<HTMLElement>(null);
+  const value = { open, setOpen, triggerRef, mode: props.mode ?? "click" };
+
+  return (
+    <PopoverContext.Provider value={value}>
+      {typeof props.children === "function"
+        ? props.children(value)
+        : props.children}
+    </PopoverContext.Provider>
+  );
+};
+
+export const usePopover = () => {
+  const context = useContext(PopoverContext);
+  if (!context) {
+    throw new Error(
+      "Popover compound components cannot be rendered outside the Popover component",
+    );
+  }
+  return context;
+};
+
+export const PopoverTrigger = (props: {
+  children: TriggerElement;
+  hover?: boolean;
+}) => {
+  const popover = usePopover();
+
+  const clickProps = {
+    onClick: () => {
+      popover.setOpen((open) => !open);
+    },
+  };
+
+  const hoverProps = {
+    onMouseEnter: () => {
+      popover.setOpen(true);
+    },
+    onMouseLeave: () => {
+      popover.setOpen(false);
+    },
+  };
+
+  return cloneElement(props.children, {
+    ...(popover.mode === "click" ? clickProps : hoverProps),
+    ref: popover.triggerRef,
+  });
+};
+
+type Horizontal = "left" | "right";
+
+export const PopoverContent = (
+  props: Omit<MuiPopoverProps, "open" | "onClose" | "anchorEl"> & {
+    horizontal?: Horizontal;
+  },
+) => {
+  const popover = usePopover();
+  const horizontal = props.horizontal ?? "left";
+  const hoverMode = popover.mode === "hover";
+
+  return (
+    <MuiPopover
+      disablePortal
+      css={(theme) => ({
+        // When it is on hover mode, and the moude is moving from the trigger to
+        // the popover, if there is any space, the popover will be closed. I
+        // found this is a limitation on how MUI structured the component. It is
+        // not a big issue for now but we can reavaluate it in the future.
+        marginTop: hoverMode ? undefined : theme.spacing(1),
+        pointerEvents: hoverMode ? "none" : undefined,
+        "& .MuiPaper-root": {
+          minWidth: theme.spacing(40),
+          fontSize: 14,
+          pointerEvents: hoverMode ? "auto" : undefined,
+        },
+      })}
+      {...horizontalProps(horizontal)}
+      {...modeProps(popover)}
+      {...props}
+      open={popover.open}
+      onClose={() => popover.setOpen(false)}
+      anchorEl={popover.triggerRef.current}
+    />
+  );
+};
+
+const modeProps = (popover: PopoverContextValue) => {
+  if (popover.mode === "hover") {
+    return {
+      onMouseEnter: () => {
+        popover.setOpen(true);
+      },
+      onMouseLeave: () => {
+        popover.setOpen(false);
+      },
+    };
+  }
+
+  return {};
+};
+
+const horizontalProps = (horizontal: Horizontal) => {
+  if (horizontal === "right") {
+    return {
+      anchorOrigin: {
+        vertical: "bottom",
+        horizontal: "right",
+      },
+      transformOrigin: {
```

Comment:
Is there a reason why the transform origin isn't also getting applied to the left case? Is it just because the transform origin already defaults to top/left?

Reply:
Yes, it is because it is the default.